### ISSUE TITLE
Clear test result when test code changed

### ISSUE THREAD

**Issue by akaroml** [microsoft/vscode-java-test#42]
Currently, the test result stays when a user changes the actual test code. This could be misleading.

Please clear the test result whenever the test code is changed.

**Comment by jdneo** [comment]
## TL;DR

Instead of clearing the results when code changes, I prefer to clear the related results before running the tests.

### Why not clear results when code changes
To clear the results when code changes, we need to listen to the code change at the client side, then let the server side to resolve the changed part and return back to the client side. Actually, things will become much more complex when:
- It contains parameterized tests
- It contains inherited classes

### Why clear the results before running the tests
- It's just the same behavior as the IntelliJ & Eclipse. After running some tests, the test reports will just show at some place, unless the user reruns it, the content of the report won't change.
- Almost no effect on the performance.

**Comment by jdneo** [comment]
Copied from @sappersix in https://github.com/redhat-developer/vscode-java/issues/944#issue-451006185

> Issue Type: Bug
> 
> When I delete a test case from a test Class and click "Run Test" in CodeLens, my test shows failure. When I open the Java Test Report, I see a list that looks like this:
> 
> ```
> case1 - Passed
> nameOfTestClase#nameOfDeletedTest - Failed  <----- There is an enty for the deleted test
> case3 - Passed
> ```
> 
> I have tried "Java: Clean the Java language server workspace" and "Java: Force Java compillation" and the problem persists.
> 
> I have to go to %userprofile%\AppData\Roaming\Code\User\workspaceStorage and delete my workspace. Once I restart VSC all my tests are clean.
> 
> Is there a setting I'm missing or something I need to do to refresh what tests VSC thinks are present?
> 
> Extension version: 0.45.0
> VS Code version: Code 1.34.0 (a622c65b2c713c890fcf4fbf07cf34049d5fe758, 2019-05-15T21:59:37.030Z)
> OS version: Windows_NT x64 10.0.17763
> 
> System Info
> Item	Value
> CPUs	Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz (12 x 2208)
> GPU Status	2d_canvas: enabled
> checker_imaging: disabled_off
> flash_3d: enabled
> flash_stage3d: enabled
> flash_stage3d_baseline: enabled
> gpu_compositing: enabled
> multiple_raster_threads: enabled_on
> native_gpu_memory_buffers: disabled_software
> rasterization: enabled
> surface_synchronization: enabled_on
> video_decode: enabled
> webgl: enabled
> webgl2: enabled
> Load (avg)	undefined
> Memory (System)	15.74GB (3.85GB free)
> Process Argv	
> Screen Reader	no
> VM	0%
